### PR TITLE
Maintenance Hatch Expansion and ID/Silicon Tweaks

### PR DESCRIPTION
**Pull request by NanakoAC**
Registering ID on pAIs and allowing them to use hatches is the main thing here, been meaning to do that for a while.

The biometric data imprinting I added because pAIs depend on comparing the card's DNA to the master before allowing registration. And also because there was no way to imprint your DNA/bloodtype/fingerprint/age/etc onto a newly made ID card, even though the ID you spawn with has these things. And also because bluespace techs don't spawn with the data on their card and it was getting in the way of my testing.

I had to move the underdoor function down a level to mob/living because spiderbots aren't silicon. 

Marked as large risk because of alteration to access.dm

changes: 
  - tweak: "Spiderbots no longer block movement."
  - tweak: "Spiderbots can now use airlock maintenance hatches."
  - rscadd: "PAIs can now have the owner's ID card scanned onto them to share access."
  - tweak: "PAIs can now use airlock maintenance hatches, but only on airlocks they have access to. Requires a scanned ID"
  - tweak: "Added new rodent speech verbs for PAIs"
  - tweak: "Positronic brains and MMIs outside of a chassis can now use ping/beep/buzz audio emotes."
  - tweak: "You can now click an ID in your hand to imprint your biometric data on it, if it's not present. This is only useful if you lose the ID you spawn with and get a new one made up."

### CODE REVIEW

File: code/game/objects/items/weapons/cards_ids.dm [hunk 1]
```diff
@@ -128,7 +128,7 @@
 	desc = "A card used to provide ID and determine access across the station."
 	icon_state = "id"
 	item_state = "card-id"
-	var/access = list()
+	var/list/access = list()
```
Comment (skull132) [inline code/game/objects/items/weapons/cards_ids.dm]: Are you certain this won't generate issues anywhere, from legacy code treating access as a string which is later converted into a list?

File: code/modules/mob/living/carbon/brain/emote.dm [hunk 2]
```diff
@@ -49,9 +49,17 @@
 			src << "You whistle."
 			message = "<B>[src]</B> whistles."
 			m_type = 2
-		if ("beep")
-			src << "You beep."
+		if("beep")
 			message = "<B>[src]</B> beeps."
+			playsound(src.loc, 'sound/machines/twobeep.ogg', 50, 0)
```
Comment (skull132) [inline code/modules/mob/living/carbon/brain/emote.dm]: This appears to be missing feedback messages to the user (src)? Even with sound ques, a text response is preferred for those who play without sound for whatever reason, surely?

**Comment by skull132** [comment]
One comment requires resolution before signing off.

File: code/modules/mob/living/carbon/brain/emote.dm (hunk 2 above)
Comment (NanakoAC) [inline code/modules/mob/living/carbon/brain/emote.dm]: There's a message. src beeps. Its displayed to everyone like any other emote.

File: code/game/objects/items/weapons/cards_ids.dm (hunk 1 above)
Comment (NanakoAC) [inline code/game/objects/items/weapons/cards_ids.dm]: Yep i'm sure. it's treated like a list everywhere.
ill fix any problems that result from this

**Comment by NanakoAC** [comment]
I'd say good to go

File: code/modules/mob/living/carbon/brain/emote.dm (hunk 2 above)
Comment (skull132) [inline code/modules/mob/living/carbon/brain/emote.dm]: Are you sure? Every other emote has a specific `src << "thing"` mark with it.

**Pull request merged**